pkg/providers/vsphere/vsphere.go

```diff
@@ -1152,11 +1156,35 @@ func (p *vsphereProvider) PostBootstrapDeleteForUpgrade(ctx context.Context) err
 	return nil
 }
 
-// PreCoreComponentsUpgrade staisfies the Provider interface.
+// PreCoreComponentsUpgrade satisfies the Provider interface.
 func (p *vsphereProvider) PreCoreComponentsUpgrade(
 	ctx context.Context,
 	cluster *types.Cluster,
 	clusterSpec *cluster.Spec,
 ) error {
+	// This is only needed for EKS-A v0.17 because UDP offloading needs to be disabled for
+	// the new Cilium version to work with VSphere clusters. Since our templates that were shipped
+	// in v0.16 releases still have UDP offloading enabled in the nodes, we must apply the daemon set
+	// to disable UDP offloading in all the nodes before upgrading Cilium.
+	// We will remove this in EKS-A release v0.18.0.
+	logger.V(4).Info("Applying ethtool daemonset to disable udp offloading")
+	return p.applyEthtoolDaemonSet(ctx, cluster, clusterSpec)
```

Comment:
can we add a guard clause and only run it for the osFamily that need it? (I think ubuntu and redhat?)